src/main/kotlin/app/revanced/patcher/apk/ApkBundle.kt

```diff
@@ -0,0 +1,156 @@
+package app.revanced.patcher.apk
+
+import app.revanced.patcher.Patcher
+import app.revanced.patcher.PatcherOptions
+import com.reandroid.apk.ApkModule
+import com.reandroid.apk.ResourceIds
+import com.reandroid.apk.xmlencoder.EncodeMaterials
+import com.reandroid.arsc.util.FrameworkTable
+import com.reandroid.arsc.value.ResConfig
+import com.reandroid.common.TableEntryStore
+import java.io.File
+
+/**
+ * An [Apk] file of type [Apk.Split].
+ *
+ * @param files A list of apk files to load.
+ */
+class ApkBundle(files: List<File>) {
+    /**
+     * The [Apk.Base] of this [ApkBundle].
+     */
+    val base: Apk.Base
+
+    /**
+     * A map containing all the [Apk.Split]s in this bundle associated by their configuration.
+     */
+    val splits: Map<String, Apk.Split>?
+
+    private fun ApkModule.isFeatureModule() = androidManifestBlock.manifestElement.let {
+        it.searchAttributeByName("isFeatureSplit")?.valueAsBoolean == true || it.searchAttributeByName("configForSplit") != null
+    }
```

Comment:
Move this below the constructor, may be part of a companion object and not an instance method of `ApkBundle`.